libexec/swiftenv-install

```diff
@@ -193,21 +193,44 @@ build_version() {
 }
 
 find_binary_url_from_api() {
-  vlog "Checking for a URL for the $VERSION on $(get_platform)."
-
-  status_code=$(curl -o /dev/null --silent --head --write-out '%{http_code}' "https://swiftenv-api.fuller.li/versions/$VERSION/binaries/$(get_platform)" || true)
+  vlog "Checking for a URL for Swift $VERSION on $(get_platform)."
+
+  case $(get_platform) in
+    'Ubuntu:16.04' )
+      SWIFT_BUILD_FOLDER="ubuntu1604"
+      SWIFT_ZIP_FILE="ubuntu16.04"
+      ;;
+    'Ubuntu:18.04' )
+      SWIFT_BUILD_FOLDER="ubuntu1804"
+      SWIFT_ZIP_FILE="ubuntu18.04"
+      ;;
+    'Ubuntu:20.04' )
+      SWIFT_BUILD_FOLDER="ubuntu2004"
+      SWIFT_ZIP_FILE="ubuntu20.04"
+      ;;
+    'CentOS Linux:7')
+      SWIFT_BUILD_FOLDER="centos7"
+      SWIFT_ZIP_FILE="centos7"
+      ;;
+    'CentOS Linux:8')
+      SWIFT_BUILD_FOLDER="centos8"
+      SWIFT_ZIP_FILE="centos8"
+      ;;
+    'Amazon Linux:2')
+      SWIFT_BUILD_FOLDER="amazonlinux2"
+      SWIFT_ZIP_FILE="amazonlinux2"
+      ;;
+    'osx')
+      url="https://swift.org/builds/swift-$VERSION-release/xcode/swift-$VERSION-RELEASE/swift-$VERSION-RELEASE-osx.pkg"
+      return
+      ;;
+    *)
+      echo "There was a problem identifying your OS $(get_platform)"
+      exit 1
+      ;;
+  esac
 
-  if [ "$status_code" = "404" ]; then
-    vlog "Did not find a binary release for $VERSION on $(get_platform)."
```

Comment:
This behaviour has changed a bit, in the past `swiftenv install` has 3 modes of operation:

- (auto) default - check if the binary exists and install that, otherwise fallback to building Swift from source 
- `--no-build` (only use binary versions)
- `--build` (build from source)

The default via this code path results in falling back to installing from source. As now the URL will always be set (even if the binary doesn't exist). This behaviour will no longer work.

Reply:
I'd suggest we perform a HEAD request and behave with 404 in this way.

Reply:
The old code path didn't set `url` (can be unset) in that case and there is a check further down in the file where it see's URL is empty and `"build"` is set to `"auto"` to then fallout.

Reply:
Done